src/test/scala/tech/cryptonomic/conseil/tezos/MetadataCachingTest.scala

```diff
@@ -0,0 +1,115 @@
+package tech.cryptonomic.conseil.tezos
+
+import cats.effect._
+import cats.effect.concurrent.MVar
+import com.rklaehn.radixtree.RadixTree
+import org.scalatest.{Matchers, OneInstancePerTest, WordSpec}
+import tech.cryptonomic.conseil.generic.chain.PlatformDiscoveryTypes.{Attribute, DataType, Entity, KeyType}
+import tech.cryptonomic.conseil.tezos.MetadataCaching._
+
+import scala.concurrent.ExecutionContext
+
+class MetadataCachingTest extends WordSpec with Matchers with OneInstancePerTest {
+
+  implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
+
+
+  val cachingStatus: MVar[IO, CachingStatus] = MVar[IO].of[CachingStatus](NotStarted).unsafeRunSync()
+  val attributesCache: MVar[IO, AttributesCache] = MVar[IO].empty[AttributesCache].unsafeRunSync()
+  val entitiesCache: MVar[IO, EntitiesCache] = MVar[IO].empty[EntitiesCache].unsafeRunSync()
+  val attributeValuesCache: MVar[IO, AttributeValuesCache] = MVar[IO].empty[AttributeValuesCache].unsafeRunSync()
+
+  val sut: MetadataCaching[IO] = new MetadataCaching[IO](cachingStatus, attributesCache, entitiesCache, attributeValuesCache)
+
+  "Metadata caching" should {
+    "get initial state of caching status" in {
+      sut.getCachingStatus.unsafeRunSync() shouldBe NotStarted
+    }
+
+    "update cache status" in {
+      sut.updateCachingStatus(InProgress).unsafeRunSync()
+
+      sut.getCachingStatus.unsafeRunSync() shouldBe InProgress
+    }
+
+    "init attributes cache" in {
+      val emptyAttributesCache: AttributesCache = Map("example" -> (0L, List()))
+      sut.putAllAttributesIntoEmptyCache(emptyAttributesCache).unsafeRunSync()
+
+      sut.getAttributes("not valid").unsafeRunSync() shouldBe None
+      sut.getAttributes("example").unsafeRunSync() shouldBe Some(0L, List())
+    }
+
+    "init entities cache" in {
+      val emptyEntitiesCache: EntitiesCache = Map("example" -> (0L, List()))
+      sut.putAllEntitiesIntoEmptyCache(emptyEntitiesCache).unsafeRunSync()
+
+      sut.getEntities("not valid").unsafeRunSync() shouldBe None
+      sut.getEntities("example").unsafeRunSync() shouldBe Some(0L, List())
+    }
+
+    "init attribute values cache" in {
+      val attributeValuesCache: AttributeValuesCache = Map("table.column" -> (0L, RadixTree[String, String]()))
+      sut.putAllAttributeValuesIntoEmptyCache(attributeValuesCache).unsafeRunSync()
+
+      sut.getAttributeValues("not valid", "not valid either").unsafeRunSync() shouldBe None
+      val (_, result) = sut.getAttributeValues("table", "column").unsafeRunSync().get
```

Comment:
You can mixin `OptionValues`in the spec to get the `.value` method to extract options and a meaningful error if the option's not there

`val (_, result) = sut.getAttributesValues(...).unsafeRunSync.value`

Reply:
ok, you're right